test/permission.validation.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const metatests = require('metatests');
+
+const {
+  setup,
+  prepareRemoteProvider,
+} = require('./fixtures/permission-validation/setup');
+
+const runTests = (test, ctx) => {
+  const testsDir = path.join(__dirname, 'permissions');
+  fs.readdir(
+    testsDir,
+    test.cbFail(testFiles => {
+      for (const testFile of testFiles) {
+        const testFilePath = path.join(testsDir, testFile);
+        const subtest = require(testFilePath);
+        subtest(test, ctx);
+      }
+    })
+  );
+};
+
+metatests.test('Permission validation test', async test => {
+  let ctx;
+
+  try {
+    ctx = await setup();
+  } catch (err) {
+    if (process.env.CI) {
+      test.fail('PostgresDB setup failed');
+    } else {
+      console.error(
+        'Cannot setup PostgresDB, skipping permission validation tests'
+      );
+    }
+    console.error(err);
+    return null;
+  }
+
+  test.beforeEach(prepareRemoteProvider(ctx.schema));
+  test.afterEach(async test => test.context.remoteProvider.close());
+
+  runTests(test, ctx);
+
+  return new Promise(resolve =>
```

Comment:
```suggestion
  await once(test)('done');
  await ctx.provider.close();
  ctx.server.close();
```

If you define:
```js
const once = emitter => event =>
  util.promisify(emitter.once).bind(emitter)(event);
```

Reply:
@nechaido since `util.promisify()` works with functions with error-first callback, this wouldn't work.